src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php

```diff
@@ -108,6 +108,38 @@ protected function getModel()
     }
 
     /**
+     * validate and save model after update or insertion.
+     * Use the reference node and tag to rename validation output for a specific node to a new offset, which makes
+     * it easier to reference specific uuids without having to use them in the frontend descriptions.
+     * @param $node reference node, to use as relative offset
+     * @return array result / validation output
+     */
+    protected function save($node = null)
+    {
+        $mdl = $this->getModel();
+        $result = array("result"=>"failed","validations" => array());
+        // perform validation
+        $valMsgs = $mdl->performValidation();
+        foreach ($valMsgs as $field => $msg) {
+            // replace absolute path to attribute for relative one at uuid.
+            if ($node != null) {
+                $fieldnm = str_replace($node->__reference, static:$internalModelName, $msg->getField());
+                $result["validations"][$fieldnm] = $msg->getMessage();
+            } else {
+                $result["validations"][$msg->getField()] = $msg->getMessage();
```

Comment:
as long as you send a $node, static:$internalModelName is prepended. looking at the shaper code, the != null could probably go away because the save isn't used to save anything else.